ios/browser/api/sync/brave_sync_internals.mm

```diff
@@ -0,0 +1,199 @@
+#include "ios/components/webui/web_ui_url_constants.h"
+#import "ios/web/js_messaging/web_view_web_state_map.h"
+#import "ios/web/public/browser_state.h"
+#import "ios/web/public/navigation/navigation_manager.h"
+#include "ios/web/public/thread/web_thread.h"
+#import "ios/web/public/web_state.h"
+#import "ios/web/public/web_state_delegate_bridge.h"
+#import "ios/web/public/web_state_observer_bridge.h"
+#import "ios/web/public/webui/web_ui_ios.h"
+#import "ios/web/web_state/ui/crw_web_controller.h"
+#import "ios/web/web_state/ui/wk_web_view_configuration_provider.h"
+#import "ios/web/web_state/web_state_impl.h"
+#import "ios/web/webui/web_ui_ios_impl.h"
+#import "net/base/mac/url_conversions.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+namespace web {
+
+WKWebView* EnsureWebViewCreatedWithConfiguration(
+    WebState* web_state,
+    WKWebViewConfiguration* configuration) {
+  WebStateImpl* impl = static_cast<WebStateImpl*>(web_state);
+  BrowserState* browser_state = impl->GetBrowserState();
+  CRWWebController* web_controller = impl->GetWebController();
+
+  WKWebViewConfigurationProvider& provider =
+      WKWebViewConfigurationProvider::FromBrowserState(browser_state);
+  provider.ResetWithWebViewConfiguration(configuration);
+
+  // |web_controller| will get the |configuration| from the |provider| to create
+  // the webView to return.
+  return [web_controller ensureWebViewCreated];
+}
+
+}  // namespace web
+
+@protocol CRWebViewPrivate<NSObject>
+- (void)setWebView:(WKWebView*)webView;
+@end
+
+@interface BraveSyncInternalsController ()<CRWWebStateDelegate, CRWWebStateObserver> {
+  ChromeBrowserState* _chromeBrowserState;
+  std::unique_ptr<web::WebState> _webState;
+  std::unique_ptr<web::WebStateObserverBridge> _webStateObserver;
+  std::unique_ptr<web::WebStateDelegateBridge> _webStateDelegate;
+  
+  WKWebView* _webView;
+  CRWWebController* _webController;
+}
+@end
+
+@implementation BraveSyncInternalsController
+- (instancetype)initWithBrowserState:(ChromeBrowserState*)mainBrowserState {
+  if ((self = [super init])) {
+    _chromeBrowserState = mainBrowserState;
+    web::WebState::CreateParams webStateCreateParams(mainBrowserState);
+    _webState = web::WebState::Create(webStateCreateParams);
+
+    _webStateObserver = std::make_unique<web::WebStateObserverBridge>(self);
+    _webState->AddObserver(_webStateObserver.get());
+
+    _webStateDelegate = std::make_unique<web::WebStateDelegateBridge>(self);
+    _webState->SetDelegate(_webStateDelegate.get());
+
+    _webController = static_cast<web::WebStateImpl*>(_webState.get())->GetWebController();
+    _webView = [_webController ensureWebViewCreated];
+
+    [self loadURL:[NSString stringWithFormat:@"%s/%s", kChromeUIScheme,
+                                             kChromeUISyncInternalsHost]];
```

Comment:
Lets move this to a lifecycle like `viewDidLoad` or `viewWillAppear` now that this is a `UIViewController` :)